presto-main/src/main/java/io/prestosql/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -208,7 +208,7 @@ public Expression visitTableScan(TableScanNode node, Void context)
         {
             Map<ColumnHandle, Symbol> assignments = ImmutableBiMap.copyOf(node.getAssignments()).inverse();
 
-            TupleDomain<ColumnHandle> predicate = metadata.getTableProperties(session, node.getTable()).getPredicate();
+            TupleDomain<ColumnHandle> predicate = node.getEnforcedConstraint();
```

Comment:
This is ok during the transition out of TableLayouts, but once that's complete, we should replace it with the call to `getTableProperties()` again. It's the responsibility of connectors to track and describe the predicate guaranteed by the TableScan. Eventually, we want to get rid of `getEnforcedConstraint` altogether. Can you add a comment that says "TODO: replace with metadata.getTableProperties() when table layouts are fully removed"?

Reply:
> It's the responsibility of connectors to track and describe the predicate guaranteed by the TableScan

It's not enough to track predicates guaranteed by table scan as this method might be called without any predicates pushed down. We should limit how many partitions we fetch or return `TupleDomain.all()` if there are too many partitions.